packages/kbn-text-based-editor/src/text_based_languages_editor.tsx

```diff
@@ -526,6 +526,13 @@ export const TextBasedLanguagesEditor = memo(function TextBasedLanguagesEditor({
     }
   }, [calculateVisibleCode, code, isCompactFocused, queryString]);
 
+  useEffect(() => {
```

Comment:
Why do we need the useEffect here? I have integrated the editor in multiple places and it works without the useEffect 🤔

Reply:
Hey @stratoula :) Here you can find a recording of the issue, basically when the editor is expanded and we pass new `query` prop it doesn't update the internal state

https://github.com/elastic/kibana/assets/5188868/6dd0d839-2738-46db-a0c7-cf1e67961117

Reply:
I see but this causes multiple rerenders every time the user types something in the editor. Maybe we cn find another solution together? I can't easily follow the code and I am not very familiar with this app but maybe we can see this together?